File: kkn_spring_boot/src/main/java/com/kkn/www/like/LikeController.java
package com.kkn.www.like;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import com.kkn.www.like.dto.LikeLoadSendDto;
import com.kkn.www.like.dto.LikeSaveDeleteSendDto;

@RestController
@RequestMapping("/like")
@CrossOrigin(origins ="*")
public class LikeController {
	@Autowired
	LikeService likeService;
	
	@PostMapping("")
	public List<Boolean> likeCheckListLoad(@RequestBody LikeLoadSendDto likeLoadSendDto) {
		return likeService.likeCheckListLoadService(likeLoadSendDto);
	}
	
	@PostMapping("/create")
	public void likeCheckListSave(@RequestBody LikeSaveDeleteSendDto likeSaveDeleteSendDto) {
		likeService.likeCheckListSaveService(likeSaveDeleteSendDto);
	}
	
	@PostMapping("/delete")
	public void likeCheckListDelete(@RequestBody LikeSaveDeleteSendDto likeSaveDeleteSendDto) {
		likeService.likeCheckListDeleteService(likeSaveDeleteSendDto);
	}
}

File: kkn_/lib/like/like_controller.dart
import 'dart:convert';

import 'package:kkn_/like/dto/like_load_response_dto.dart';
import 'package:kkn_/like/dto/like_load_send_dto.dart';
import 'package:kkn_/like/dto/like_save_delete_send_dto.dart';
import 'package:kkn_/server_connector.dart';

class LikeController {
  Future<LikeLoadResponseDto> likeCheckListLoad(
      LikeLoadSendDto likeLoadSendDto) async {
    ServerConnector serverConnector = ServerConnector();

    LikeLoadResponseDto likeLoadResponseDto = LikeLoadResponseDto("", []);

    try {
      String response = await serverConnector.sendProcess(
          "/like", json.encode(likeLoadSendDto.toJson()));

      likeLoadResponseDto.likeCheckList = json.decode(response);
    } catch (exception) {
      likeLoadResponseDto.errorMessage =
          serverConnector.serverExceptionProcess(exception);
    }

    return likeLoadResponseDto;
  }

  Future<String> likeCheckListSave(
      LikeSaveDeleteSendDto likeSaveSendDto) async {
    ServerConnector serverConnector = ServerConnector();

    try {
      await serverConnector.sendProcess(
          "/like/create", json.encode(likeSaveSendDto.toJson()));

      return "";
    } catch (exception) {
      return serverConnector.serverExceptionProcess(exception);
    }
  }

  Future<String> likeCheckListDelete(
      LikeSaveDeleteSendDto likeSaveSendDto) async {
    ServerConnector serverConnector = ServerConnector();

    try {
      await serverConnector.sendProcess(
          "/like/delete", json.encode(likeSaveSendDto.toJson()));

      return "";
    } catch (exception) {
      return serverConnector.serverExceptionProcess(exception);
    }
  }
}
